File: semaine2/Dizainier/Dizainier/ViewController.m
//
//  ViewController.m
//  Dizainier
//
//  Created by Cao Sang DOAN on 22/09/15.
//  Copyright (c) 2015 Cao Sang DOAN. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _stepper.minimumValue = 0;
    _stepper.maximumValue = 99;
    _stepper.stepValue = 1;
    
    _slider.minimumValue = 0;
    _slider.maximumValue = 99;
    [_slider setValue:0];
    
    [_geekSwitcher setOn:NO];
    
    [_numberLabel setTextColor:UIColor.blueColor];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)dealloc
{
    [_stepper release];
    [_geekLabel release];
    [_geekSwitcher release];
    [_dizaine release];
    [_unite release];
    [_numberLabel release];
    [_slider release];

    [super dealloc];
}

- (IBAction)stepperAction:(UIStepper *)sender {
   
    int value = [_stepper value];
    _dix = value / 10;
    _unit = value % 10;
    
    [_slider setValue:value];
    _dizaine.selectedSegmentIndex = _dix;
    _unite.selectedSegmentIndex = _unit; 
    
    if ([_geekSwitcher isOn])
    {
        
        _numberLabel.text = [NSString stringWithFormat:@"%02x", value];
    }
    else
    {
        _numberLabel.text = [NSString stringWithFormat:@"%d", value];
    }
    
    int tmp = value % 10;
    
    switch (tmp) {
        case 0:
            [_numberLabel setTextColor:UIColor.blueColor];
            break;
        case 1:
            [_numberLabel setTextColor:UIColor.redColor];
            break;
        case 2:
            [_numberLabel setTextColor:UIColor.greenColor];
            break;
        case 3:
            [_numberLabel setTextColor:UIColor.grayColor];
            break;
        case 4:
            [_numberLabel setTextColor:UIColor.yellowColor];
            break;
        case 5:
            [_numberLabel setTextColor:UIColor.cyanColor];
            break;
        case 6:
            [_numberLabel setTextColor:UIColor.blackColor];
            break;
        case 7:
            [_numberLabel setTextColor:UIColor.purpleColor];
            break;
        case 8:
            [_numberLabel setTextColor:UIColor.magentaColor];
            break;
        case 9:
            [_numberLabel setTextColor:UIColor.orangeColor];
            break;
    }
}

- (IBAction)geekAction:(UISwitch *)sender {
    if ([_geekSwitcher isOn])
    {
        _geekLabel.text = @"Geek";
        _numberLabel.text = [NSString stringWithFormat:@"%02x", _dix*10+_unit];
    }
    else
    {
        _geekLabel.text = @"Normal";
        _numberLabel.text = [NSString stringWithFormat:@"%d", _dix*10+_unit];
    }
}

- (IBAction)dizainAction:(UISegmentedControl *)sender {
    _dix = (int)_dizaine.selectedSegmentIndex;
    
    int value = _dix * 10 + _unit;
    
    [_slider setValue:value];
    [_stepper setValue:value];
    
    if ([_geekSwitcher isOn])
    {
        _numberLabel.text = [NSString stringWithFormat:@"%02x", value];
    }
    else
    {
        _numberLabel.text = [NSString stringWithFormat:@"%d", value];
    }
    
    int tmp = value % 10;
    
    switch (tmp) {
        case 0:
            [_numberLabel setTextColor:UIColor.blueColor];
            break;
        case 1:
            [_numberLabel setTextColor:UIColor.redColor];
            break;
        case 2:
            [_numberLabel setTextColor:UIColor.greenColor];
            break;
        case 3:
            [_numberLabel setTextColor:UIColor.grayColor];
            break;
        case 4:
            [_numberLabel setTextColor:UIColor.yellowColor];
            break;
        case 5:
            [_numberLabel setTextColor:UIColor.cyanColor];
            break;
        case 6:
            [_numberLabel setTextColor:UIColor.blackColor];
            break;
        case 7:
            [_numberLabel setTextColor:UIColor.purpleColor];
            break;
        case 8:
            [_numberLabel setTextColor:UIColor.magentaColor];
            break;
        case 9:
            [_numberLabel setTextColor:UIColor.orangeColor];
            break;
    }

}

- (IBAction)uniteAction:(UISegmentedControl *)sender {
    _unit = (int)_unite.selectedSegmentIndex;
    
    int value = _dix * 10 + _unit;
    
    [_slider setValue:value];
    [_stepper setValue:value];
    
    if ([_geekSwitcher isOn])
    {
        _numberLabel.text = [NSString stringWithFormat:@"%02x", value];
    }
    else
    {
        _numberLabel.text = [NSString stringWithFormat:@"%d", value];
    }
    
    int tmp = value % 10;
    
    switch (tmp) {
        case 0:
            [_numberLabel setTextColor:UIColor.blueColor];
            break;
        case 1:
            [_numberLabel setTextColor:UIColor.redColor];
            break;
        case 2:
            [_numberLabel setTextColor:UIColor.greenColor];
            break;
        case 3:
            [_numberLabel setTextColor:UIColor.grayColor];
            break;
        case 4:
            [_numberLabel setTextColor:UIColor.yellowColor];
            break;
        case 5:
            [_numberLabel setTextColor:UIColor.cyanColor];
            break;
        case 6:
            [_numberLabel setTextColor:UIColor.blackColor];
            break;
        case 7:
            [_numberLabel setTextColor:UIColor.purpleColor];
            break;
        case 8:
            [_numberLabel setTextColor:UIColor.magentaColor];
            break;
        case 9:
            [_numberLabel setTextColor:UIColor.orangeColor];
            break;
    }

}

- (IBAction)sliderAction:(UISlider *)sender {
    int value = [_slider value];
    _dix = value / 10;
    _unit = value % 10;
    
    [_stepper setValue:value];
    _dizaine.selectedSegmentIndex = _dix;
    _unite.selectedSegmentIndex = _unit; 
    
    if ([_geekSwitcher isOn])
    {
        _numberLabel.text = [NSString stringWithFormat:@"%02x", value];
    }
    else
    {
        _numberLabel.text = [NSString stringWithFormat:@"%d", value];
    }
    
    int tmp = value % 10;
    
    switch (tmp) {
        case 0:
            [_numberLabel setTextColor:UIColor.blueColor];
            break;
        case 1:
            [_numberLabel setTextColor:UIColor.redColor];
            break;
        case 2:
            [_numberLabel setTextColor:UIColor.greenColor];
            break;
        case 3:
            [_numberLabel setTextColor:UIColor.grayColor];
            break;
        case 4:
            [_numberLabel setTextColor:UIColor.yellowColor];
            break;
        case 5:
            [_numberLabel setTextColor:UIColor.cyanColor];
            break;
        case 6:
            [_numberLabel setTextColor:UIColor.blackColor];
            break;
        case 7:
            [_numberLabel setTextColor:UIColor.purpleColor];
            break;
        case 8:
            [_numberLabel setTextColor:UIColor.magentaColor];
            break;
        case 9:
            [_numberLabel setTextColor:UIColor.orangeColor];
            break;
    }

}

- (IBAction)resetAction:(UIButton *)sender {
    
    _dix = 0;
    _unit = 0;
    _stepper.value = 0;
    
    [_slider setValue:0];
    
    [_geekSwitcher setOn:NO];
    
    [_numberLabel setTextColor:UIColor.blueColor];
    
    _dizaine.selectedSegmentIndex = _dix;
    _unite.selectedSegmentIndex = _unit;
    
    _numberLabel.text = @"0";
    _geekLabel.text = @"Normal";
}
@end

File: semaine3/Dizainier_Swift/Dizainier_Swift/ViewController.swift
//
//  ViewController.swift
//  Dizainier_Swift
//
//  Created by Cao Sang DOAN on 25/09/15.
//  Copyright (c) 2015 Cao Sang DOAN. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var geekLabel: UILabel!
    @IBOutlet weak var stepper: UIStepper!
    @IBOutlet weak var dizaines: UISegmentedControl!
    @IBOutlet weak var unites: UISegmentedControl!
    @IBOutlet weak var numberLabel: UILabel!
    @IBOutlet weak var slider: UISlider!
    @IBOutlet weak var reset: UIButton!
    @IBOutlet weak var switchButton: UISwitch!
    
    var value = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        slider.minimumValue = 0
        slider.maximumValue = 99
        
        switchButton.setOn(false, animated: false)
        
        stepper.minimumValue = 0
        stepper.maximumValue = 99
        stepper.stepValue = 1
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func changeMode(sender: UISwitch) {
        affichage();
    }

    @IBAction func changeStepper(sender: UIStepper) {
        value = Int(stepper.value)
        affichage()
    }
    
    @IBAction func changeSegment(sender: UISegmentedControl) {
        value = Int(dizaines.selectedSegmentIndex * 10 + unites.selectedSegmentIndex)
        affichage()
    }

    
    @IBAction func changeSlider(sender: UISlider) {
        value = Int(slider.value)
        affichage()
    }
    
    @IBAction func changeReset(sender: UIButton) {
        value = 0
        affichage()
    }
    
    func affichage()
    {
        stepper.value = Double(value)
        dizaines.selectedSegmentIndex = value / 10
        unites.selectedSegmentIndex = value % 10
        slider.value = Float(value)
        
        if (switchButton.on)
        {
            geekLabel.text = "Geek"
            numberLabel.text = String(value, radix: 16)
        }
        else
        {
            geekLabel.text = "Normal"
            numberLabel.text = String(value, radix: 10)
        }
        
        let tmp = value % 10;
        
        switch (tmp) {
        case 1:
            numberLabel.textColor = UIColor.redColor()
            break
        case 2:
            numberLabel.textColor = UIColor.greenColor()
            break
        case 3:
            numberLabel.textColor = UIColor.grayColor()
            break
        case 4:
            numberLabel.textColor = UIColor.yellowColor()
            break
        case 5:
            numberLabel.textColor = UIColor.cyanColor()
            break
        case 6:
            numberLabel.textColor = UIColor.blackColor()
            break
        case 7:
            numberLabel.textColor = UIColor.purpleColor()
            break
        case 8:
            numberLabel.textColor = UIColor.magentaColor()
            break
        case 9:
            numberLabel.textColor = UIColor.orangeColor()
            break
        default:
            numberLabel.textColor = UIColor.blueColor()
            break
        }

    }
    
    
    
}

